server/storage/wal/version.go

```diff
@@ -53,32 +53,69 @@ func (w *walVersion) MinimalEtcdVersion() *semver.Version {
 func MinimalEtcdVersion(ents []raftpb.Entry) *semver.Version {
 	var maxVer *semver.Version
 	for _, ent := range ents {
-		maxVer = maxVersion(maxVer, etcdVersionFromEntry(ent))
+		err := visitEntry(ent, func(path protoreflect.FullName, ver *semver.Version) error {
+			maxVer = maxVersion(maxVer, ver)
+			return nil
+		})
+		if err != nil {
+			panic(err)
+		}
 	}
 	return maxVer
 }
 
-func etcdVersionFromEntry(ent raftpb.Entry) *semver.Version {
-	msgVer := etcdVersionFromMessage(proto.MessageReflect(&ent))
-	dataVer := etcdVersionFromData(ent.Type, ent.Data)
-	return maxVersion(msgVer, dataVer)
+type Visitor func(path protoreflect.FullName, ver *semver.Version) error
+
+func VisitFileDescriptor(file protoreflect.FileDescriptor, visitor Visitor) error {
```

Comment:
Please document that this method calls visitor on each field and enum value with the 
version coming from the configured option or nil if unset.

Reply:
Done